app/components/Session/NewSessionForm.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import { Button, Switch, Input, Modal, Form, Icon, Row, Col } from 'antd';
+import { remote } from 'electron';
+import FormattedCaps from './FormattedCaps';
+
+const { dialog } = remote;
+const FormItem = Form.Item;
+
+export default class NewSessionForm extends Component {
+
+  getLocalFilePath (success) {
+    dialog.showOpenDialog((filepath) => {
+      if (filepath) {
+        success(filepath);
+      }
+    });
+  }
+
+  getCapsObject () {
+    const { caps } = this.props;
+    let capsObject = {};
+    caps.forEach((cap) => capsObject[cap.name] = cap.value);
+    return capsObject;
+  }
+
+  render () {
+    const { setCapabilityParam, caps, addCapability, removeCapability, saveSession, hideSaveAsModal, saveAsText, showSaveAsModal, setSaveAsText } = this.props;
+
+    return <div>
+      <h4>Desired Capabilities</h4> 
+      <Row> 
+      <Col span={12}>
+        <Form inline>
+          {caps.map((cap, index) => {
+            const buttonAfter = <Icon style={{cursor: 'pointer'}} type="file" onClick={() => this.getLocalFilePath((filepath) => setCapabilityParam(index, 'value', filepath))} />;
+            return <div key={index} style={{marginBottom: '1em'}}>
+              <FormItem>
+                <Input placeholder='Name' value={cap.name} onChange={(e) => setCapabilityParam(index, 'name', e.target.value)}/>
+              </FormItem>
+              <FormItem>
+                <select onChange={(e) => setCapabilityParam(index, 'type', e.target.value)} value={cap.type}>
+                  <option value='text'>text</option>
+                  <option value='boolean'>boolean</option>
+                  <option value='number'>number</option>
+                  <option value='json_object'>JSON object</option>
+                  <option value='file'>filepath</option>
+                </select>
+              </FormItem>
+              <FormItem>
+                {
+                  function () {
```

Comment:
feels like it would be cleaner to just factor this out and call it?

```js
<FormItem>
  {this.getCapControl(cap)}
</FormItem>
```

(might need to also pass in your `buttonAfter`, or build that inside this helper method instead